src/Components/test/E2ETestMigration/Tests/BindTest.cs

```diff
@@ -0,0 +1,1367 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using BasicTestApp;
+using Microsoft.AspNetCore.BrowserTesting;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.Testing;
+using PlaywrightSharp;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.Tests
+{
+    public class BindTest : ServerTestBase<ToggleExecutionModeServerFixture<Program>>
+    {
+        public BindTest(
+            ToggleExecutionModeServerFixture<Program> serverFixture,
+            ITestOutputHelper output)
+            : base(serverFixture, output)
+        {
+        }
+
+        protected override Type TestComponent { get; } = typeof(BindCasesComponent);
+
+        protected override async Task InitializeCoreAsync(TestContext context)
+        {
+            await base.InitializeCoreAsync(context);
+
+            // On WebAssembly, page reloads are expensive so skip if possible
+            //Navigate(ServerPathBase, noReload: _serverFixture.ExecutionMode == ExecutionMode.Client);
+        }
+
+        protected async Task<string> GetInputValue(string selector)
+            => await TestPage.EvalOnSelectorAsync<string>(selector, "e => e.value");
+
+        [Theory]
+        [InlineData(BrowserKind.Chromium)]
+        [InlineData(BrowserKind.Firefox)]
+        [InlineData(BrowserKind.Webkit)]
+        // NOTE: BrowserKind argument must be first
+        public async Task CanBindTextbox_InitiallyBlank(BrowserKind browserKind)
+        {
+            if (ShouldSkip(browserKind)) 
+            {
+                return;
+            }
+
+            var target = await TestPage.QuerySelectorAsync("#textbox-initially-blank");
+            Assert.Equal(string.Empty, await GetInputValue("#textbox-initially-blank"));
+            Assert.Equal(string.Empty, await TestPage.GetInnerTextAsync("#textbox-initially-blank-value"));
+            Assert.Equal(string.Empty, await GetInputValue("#textbox-initially-blank-mirror"));
+
+            // Modify target; verify value is updated and that textboxes linked to the same data are updated
+            await target.TypeAsync("Changed value");
+            // Doesn't update until change event
```

Comment:
This is interesting. How does Playwright know this event is what we want it to wait for?

Also reading [the docs on this API](https://playwrightsharp.dev/api/PlaywrightSharp.ElementHandle.html#PlaywrightSharp_ElementHandle_GetInnerTextAsync_System_Nullable_System_Int32__), it's not clear to me how we should know that's what it does. Could you clarify where we get this information?

Reply:
This took some manual playing around with to get work, I initially tried the wrong method which triggered the change update, this api simulates typing in the textbox, so its not until you press tab or enter that the browser seems to trigger the change event, so this test types some characters and verifies that things don't change before, the tab key is sent using PressAsync which causes things to update